### ISSUE TITLE
[Question] Video input

### ISSUE THREAD

**Issue by OpenSourceAnarchist** [HopLee6/RRIN#3]
Hi,

Instead of 2 images, I was wondering what the best way is to have RRIN output interpolated frames for a video. If I convert the video to a sequence of images, should I modify demo.py to loop through the images and pick 2 at a time, or is there a more efficient way to handle videos/image sequences?

Thanks!

**Comment by HopLee6** [comment]
A more efficient way would be processing the image sequence in batch-wise. In my demo, the batch size is 1 because there is only one image pair. You can adapt the code to a larger batch size (according to your GPU) and even to multi-step interpolation.

**Comment by xiaozhi2015** [comment]
Have you load the pretrained model successfully? @OpenSourceAnarchist

**Comment by OpenSourceAnarchist** [comment]
@xiaozhi2015 Sorry for not responding, I have not attempted this yet. I was finishing up the semester and I have been looking at sniklaus's newest motion interpolation method https://github.com/sniklaus/softmax-splatting. Having the same situation with needing to loop through 2 images at a time! If I implement it there, I'll copy the code back over to RRIN and post it here.

**Comment by xiaozhi2015** [comment]
@OpenSourceAnarchist. Problem has been solved: #4.
I also followed up softmax-splatting program, hope for further communication.

**Comment by OpenSourceAnarchist** [comment]
Glad the issue was resolved!

As for video input, I ended up doing a very naïve single batch approach because I have never tried to modify any NN code and have no idea where to begin with that. I work on simple cli applications in Nim and sometimes Rust, never Python.

I used ffmpeg to convert my video file to a sequence of images (%d). Then I just made a counter start at 1 and, while that var is less than the number of images:
`im1_path = 'data/' + str(imageNum) + '.png'`
`im2_path = 'data/' + str(imageNum + 1) + '.png'`

I'm running it on Google Colab in a few minutes, if it's too slow I'll look into other ways to do this. I can always use AviSynth+ and mvtools2 :)
